src/pybind/mgr/cephadm/module.py

```diff
@@ -1456,35 +1456,68 @@ def add_host(self, spec: HostSpec) -> str:
         return self._add_host(spec)
 
     @handle_orch_error
-    def remove_host(self, host):
-        # type: (str) -> str
+    def remove_host(self, host: str, force: bool = False, offline: bool = False) -> str:
         """
         Remove a host from orchestrator management.
 
         :param host: host name
+        :param force: bypass runnning daemons check
+        :param offline: remove offline host
         """
-        # Verify if it is possible to remove the host safely
-        daemons = self.cache.get_daemons_by_host(host)
-        if daemons:
-            self.log.warning(f"Blocked {host} removal. Daemons running: {daemons}")
 
-            daemons_table = ""
-            daemons_table += "{:<20} {:<15}\n".format("type", "id")
-            daemons_table += "{:<20} {:<15}\n".format("-" * 20, "-" * 15)
+        # check if offline
+        if not offline:
+            offline = host in self.offline_hosts or self.inventory._inventory[host].get(
+                'status', '').lower() == 'offline'
+
+        # Verify if it is possible to remove the host safely
+        if not force and not offline:
+            daemons = self.cache.get_daemons_by_host(host)
+            if daemons:
+                self.log.warning(f"Blocked {host} removal. Daemons running: {daemons}")
+
+                daemons_table = ""
+                daemons_table += "{:<20} {:<15}\n".format("type", "id")
+                daemons_table += "{:<20} {:<15}\n".format("-" * 20, "-" * 15)
+                for d in daemons:
+                    daemons_table += "{:<20} {:<15}\n".format(d.daemon_type, d.daemon_id)
+
+                return "Not allowed to remove %s from cluster. " \
+                    "The following daemons are running in the host:" \
+                    "\n%s\nPlease run 'ceph orch host drain %s' to remove daemons from host" % (
+                        host, daemons_table, host)
+
+        if offline and not force:
+            return "Host is offline, please use --force to remove this host."
+
+        if offline and force:
+            offline = True
+            daemons = self.cache.get_daemons_by_host(host)
             for d in daemons:
-                daemons_table += "{:<20} {:<15}\n".format(d.daemon_type, d.daemon_id)
+                self.log.info(f"removing: {d.service_name()}{d.daemon_id}")
```

Comment:
```suggestion
                self.log.info(f'removing: {d.name()}')
```
Unless I'm misunderstanding what's getting logged here, can we just use the name of the daemon?